src/twisted/trial/_asynctest.py

```diff
@@ -131,43 +131,42 @@ def onTimeout(d):
     def __call__(self, *args, **kwargs):
         return self.run(*args, **kwargs)
 
-    def deferSetUp(self, ignored, result):
-        d = self._run("setUp", result)
-        d.addCallbacks(
-            self.deferTestMethod,
-            self._ebDeferSetUp,
-            callbackArgs=(result,),
-            errbackArgs=(result,),
-        )
-        return d
+    async def deferSetUp(self, result):
+        self._runNeedsTearDown = False
+        try:
+            try:
+                await self.deferSetUpNoCleanup(result)
+            finally:
+                await self.deferRunCleanups(None, result)
+        finally:
+            if self._runNeedsTearDown:
+                await self.deferTearDown(None, result)
```

Comment:
To simplify the code
```suggestion
                await self._deferTearDown(result)
```

Reply:
Fixed now.